engine/src/org/pentaho/di/trans/steps/transexecutor/TransExecutorMeta.java

```diff
@@ -608,120 +599,7 @@ public void getFields( RowMetaInterface inputRowMeta, String name, RowMetaInterf
-        } catch ( Exception e ) {
-          throw new KettleException( BaseMessages.getString( PKG, "TransExecutorMeta.Exception.UnableToLoadTrans" ),
-              e );
-        }
-        break;
-
-      case REPOSITORY_BY_NAME:
-        String realTransname = tmpSpace.environmentSubstitute( executorMeta.getTransName() );
-        String realDirectory = tmpSpace.environmentSubstitute( executorMeta.getDirectoryPath() );
-
-        if ( rep != null ) {
-          if ( !Utils.isEmpty( realTransname ) && !Utils.isEmpty( realDirectory ) ) {
-            realDirectory = r.normalizeSlashes( realDirectory );
-            RepositoryDirectoryInterface repdir = rep.findDirectory( realDirectory );
-            if ( repdir != null ) {
-              try {
-                // reads the last revision in the repository...
-                //
-                mappingTransMeta = rep.loadTransformation( realTransname, repdir, null, true, null ); // TODO: FIXME:
-                                                                                                      // pass
-                // in metaStore to
-                // repository?
-
-                LogChannel.GENERAL.logDetailed( "Loading transformation from repository", "Executor transformation ["
-                    + realTransname + "] was loaded from the repository" );
-              } catch ( Exception e ) {
-                throw new KettleException( "Unable to load transformation [" + realTransname + "]", e );
-              }
-            }
-          }
-        } else {
-          // rep is null, let's try loading by filename
-          try {
-            mappingTransMeta =
-                new TransMeta( realDirectory + "/" + realTransname, metaStore, rep, true, tmpSpace, null );
-          } catch ( KettleException ke ) {
-            try {
-              // add .ktr extension and try again
-              mappingTransMeta =
-                  new TransMeta( realDirectory + "/" + realTransname + "." + Const.STRING_TRANS_DEFAULT_EXT, metaStore,
-                      rep, true, tmpSpace, null );
-            } catch ( KettleException ke2 ) {
-              throw new KettleException( BaseMessages.getString( PKG, "TransExecutorMeta.Exception.UnableToLoadTrans",
-                  realTransname ) + realDirectory );
-            }
-          }
-        }
-        break;
-
-      case REPOSITORY_BY_REFERENCE:
-        // Read the last revision by reference...
-        mappingTransMeta = rep.loadTransformation( executorMeta.getTransObjectId(), null );
-        break;
-      default:
-        break;
-    }
-
-    // Pass some important information to the mapping transformation metadata:
-    //
-    mappingTransMeta.copyVariablesFrom( space );
-    mappingTransMeta.setRepository( rep );
-    mappingTransMeta.setMetaStore( metaStore );
-    mappingTransMeta.setFilename( mappingTransMeta.getFilename() );
-
-    return mappingTransMeta;
+    return loadMappingMeta( executorMeta, rep, null, space );
```

Comment:
@mbatchelor, yes you are right. But as far as I understand code above makes the same thing like was done by  Mike here: https://github.com/pentaho/pentaho-kettle/pull/1664/files#diff-e56f34b7521e57126c305cd757b1c332R633. I have moved it all to one place in order to avoid duplications. lets wait for @mdamour1976, to be sure that I did not miss something.